firebase-ml-modeldownloader/src/test/java/com/google/firebase/ml/modeldownloader/FirebaseModelDownloaderTest.java

```diff
@@ -51,14 +58,28 @@
   CustomModel CUSTOM_MODEL = new CustomModel(MODEL_NAME, 0, 100, MODEL_HASH);
 
   FirebaseModelDownloader firebaseModelDownloader;
-  SharedPreferencesUtil mockPrefs = mock(SharedPreferencesUtil.class);
+  @Mock SharedPreferencesUtil mockPrefs;
+
+  ExecutorService executor;
 
   @Before
   public void setUp() {
+    MockitoAnnotations.initMocks(this);
     FirebaseApp.clearInstancesForTest();
     // default app
     FirebaseApp.initializeApp(ApplicationProvider.getApplicationContext(), FIREBASE_OPTIONS);
-    firebaseModelDownloader = new FirebaseModelDownloader(FIREBASE_OPTIONS, mockPrefs);
+
+    executor = new ThreadPoolExecutor(0, 1, 30L, TimeUnit.SECONDS, new LinkedBlockingQueue<>());
```

Comment:
You could use a singleThread executor like in

https://github.com/firebase/firebase-android-sdk/blob/fe51be95316ad7e72ba0ecd98776415ff9109b4e/firebase-messaging/src/test/java/com/google/firebase/messaging/ImageDownloadRoboTest.java#L53

That way the test shouldn't finish before running the tasks.

Reply:
Done